llvm/lib/ObjectYAML/GOFFYAML.cpp

```diff
@@ -23,6 +23,224 @@ Object::Object() {}
 
 namespace yaml {
 
+void ScalarEnumerationTraits<GOFFYAML::GOFF_ESDSYMBOLTYPE>::enumeration(
+    IO &IO, GOFFYAML::GOFF_ESDSYMBOLTYPE &Value) {
+#define ECase(X) IO.enumCase(Value, #X, GOFF::X)
+  ECase(ESD_ST_SectionDefinition);
+  ECase(ESD_ST_ElementDefinition);
+  ECase(ESD_ST_LabelDefinition);
+  ECase(ESD_ST_PartReference);
+  ECase(ESD_ST_ExternalReference);
+#undef ECase
```

Comment:
Given that this macro is identical throughout these `enumeration` functions, I'd be okay if you wanted to define the macro before the first `enumeration` and undef it at the end of the `enumeration` functions.